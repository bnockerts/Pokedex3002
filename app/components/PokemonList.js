import React, { Component } from 'react';
import {
  AppRegistry,
  ListView,
  StyleSheet,
  Text,
  TouchableHighlight,
  View
} from 'react-native';

async function getPokemon() {
  try {
    let response = await fetch('https://pokeapi.co/api/v2/pokemon/?limit=50');
    let responseJson = await response.json();
    // let result = [ { url: 'https://pokeapi.co/api/v2/pokemon/1/',
	  //   name: 'bulbasaur' },
	  // { url: 'https://pokeapi.co/api/v2/pokemon/2/', name: 'ivysaur' },
	  // { url: 'https://pokeapi.co/api/v2/pokemon/3/',
	  //   name: 'venusaur' },
	  // { url: 'https://pokeapi.co/api/v2/pokemon/4/',
	  //   name: 'charmander' }, ];

    const indicator = 'pokemon/';
    return responseJson.results.map(pokemon => {
      pokemon.id = pokemon.url.substring(pokemon.url.indexOf(indicator) + indicator.length, pokemon.url.lastIndexOf('/'));
      pokemon.name = capitalizeFirstLetter(pokemon.name);
      return pokemon;
    });
    // return result;
  } catch(err) {
    console.error(err);
  }
}

function capitalizeFirstLetter(string) {
  return string.charAt(0).toUpperCase() + string.slice(1);
}

class PokemonList extends Component {
  constructor() {
    super();
    const dataSource = new ListView.DataSource({rowHasChanged: (r1, r2) => r1.name !== r2.name});
    this.state = {
      dataSource: dataSource.cloneWithRows([])
    }
  }

  componentDidMount() {
    getPokemon().then(pokemon => {
      const dataSource = new ListView.DataSource({rowHasChanged: (r1, r2) => r1.name !== r2.name});
      this.setState({
        dataSource: dataSource.cloneWithRows(pokemon)
      });
    });
  }

  renderRow = (rowData, sectionId, rowId) => {
    return (
      <TouchableHighlight underlayColor='#bada55' onPress={this._onPressButton}>
        <View>
          <View style={styles.rowContainer}>
            <View style={styles.textContainer}>
              <Text style={styles.id}>{rowData.id}</Text>
              <Text style={styles.name} numberOfLines={1}>{rowData.name}</Text>
            </View>
          </View>
          <View style={styles.separator}/>
        </View>
      </TouchableHighlight>
    )
  }

  render() {
    return (
      <ListView
        dataSource={this.state.dataSource}
        renderRow={this.renderRow}
      />
    );
  }
}

const styles = StyleSheet.create({
  thumb: {
    width: 80,
    height: 80,
    marginRight: 10
  },
  textContainer: {
    flex: 1
  },
  separator: {
    height: 1,
    backgroundColor: '#ddd'
  },
  id: {
    fontSize: 25,
    fontWeight: 'bold',
    color: '#48BBEC'
  },
  name: {
    fontSize: 20,
    color: '#656565'
  },
  rowContainer: {
    flexDirection: 'row',
    padding: 10
  }
});

export default PokemonList;