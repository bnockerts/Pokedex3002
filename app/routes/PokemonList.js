import React, { Component } from 'react';
import Spinner from '../components/Spinner';
import PokemonDetail from './PokemonDetail';
import {
  AppRegistry,
  ListView,
  StyleSheet,
  Text,
  TouchableHighlight,
  View
} from 'react-native';

const API_URL = 'https://pokeapi.co/api/v2/pokemon/?limit=50';

async function getPokemon() {
  try {
    let response = await fetch(API_URL);
    let responseJson = await response.json();

    const indicator = 'pokemon/';
    return responseJson.results.map(pokemon => {
      pokemon.id = pokemon.url.substring(pokemon.url.indexOf(indicator) + indicator.length, pokemon.url.lastIndexOf('/'));
      pokemon.name = capitalizeFirstLetter(pokemon.name);
      return pokemon;
    });
  } catch(err) {
    console.error(err);
  }
}

function capitalizeFirstLetter(string) {
  return string.charAt(0).toUpperCase() + string.slice(1);
}

class PokemonList extends Component {
  constructor() {
    super();

    const dataSource = new ListView.DataSource({rowHasChanged: (r1, r2) => r1.name !== r2.name});
    this.state = {
      isLoading: true,
      dataSource: dataSource.cloneWithRows([])
    };
  }

  componentDidMount() {
    getPokemon()
      .then(pokemon => {
        this.setState({
          dataSource: this.state.dataSource.cloneWithRows(pokemon)
        });
      }).finally(() => {
        this.setState({
          isLoading: false
        });
      });
  }

  renderRow = (rowData, sectionId, rowId) => {
    return (
      <TouchableHighlight underlayColor='#bada55' onPress={() => this.rowPressed(rowData)}>
        <View>
          <View style={styles.rowContainer}>
            <View style={styles.container}>
              <Text style={styles.id}>{rowData.id}</Text>
              <Text style={styles.name} numberOfLines={1}>{rowData.name}</Text>
            </View>
          </View>
          <View style={styles.separator}/>
        </View>
      </TouchableHighlight>
    );
  }

  rowPressed = (pokemon) => {
    this.props.navigator.push({
      id: 'detail',
      pokemon
    });
  }

  render() {
    if (this.state.isLoading) {
      return (
        <Spinner isLoading={this.state.isLoading} />
      );
    }

    return (
      <ListView
        dataSource={this.state.dataSource}
        renderRow={this.renderRow}
      />
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  separator: {
    height: 1,
    backgroundColor: '#ddd'
  },
  id: {
    fontSize: 25,
    fontWeight: 'bold',
    color: '#48BBEC'
  },
  name: {
    fontSize: 20,
    color: '#656565'
  },
  rowContainer: {
    flexDirection: 'row',
    padding: 10
  },
  spinner: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  }
});

export default PokemonList;
