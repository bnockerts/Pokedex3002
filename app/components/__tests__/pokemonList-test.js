import React from 'react';
import renderer from 'react-test-renderer';
import PokemonList from '../PokemonList';
import { formattedPokemonListData } from '../../utils';

it('renders a pokemonList', () => {
  const component = renderer.create(
    <PokemonList
      onRowPress={jest.fn}
      onValueChange={jest.fn}
      pokemon={formattedPokemonListData}
    />
  );

  expect(component).toMatchSnapshot();
});
