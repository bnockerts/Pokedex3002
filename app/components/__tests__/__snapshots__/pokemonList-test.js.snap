exports[`test renders a pokemonList 1`] = `
<ScrollView
  dataSource={
    ListViewDataSource {
      "items": 50,
    }
  }
  initialListSize={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={1000}
  onKeyboardDidHide={undefined}
  onKeyboardDidShow={undefined}
  onKeyboardWillHide={undefined}
  onKeyboardWillShow={undefined}
  onLayout={[Function]}
  onScroll={[Function]}
  pageSize={1}
  removeClippedSubviews={true}
  renderRow={[Function]}
  scrollEventThrottle={50}
  scrollRenderAheadDistance={1000}
  stickyHeaderIndices={Array []}>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "padding": 10,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#48BBEC",
                "fontSize": 25,
                "fontWeight": "bold",
              }
            }>
            1
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#656565",
                "fontSize": 20,
              }
            }>
            Bulbasaur
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                undefined,
              ]
            }
            value={false} />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ddd",
            "height": 1,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "padding": 10,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#48BBEC",
                "fontSize": 25,
                "fontWeight": "bold",
              }
            }>
            2
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#656565",
                "fontSize": 20,
              }
            }>
            Ivysaur
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                undefined,
              ]
            }
            value={false} />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ddd",
            "height": 1,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "padding": 10,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#48BBEC",
                "fontSize": 25,
                "fontWeight": "bold",
              }
            }>
            3
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#656565",
                "fontSize": 20,
              }
            }>
            Venusaur
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                undefined,
              ]
            }
            value={false} />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ddd",
            "height": 1,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "padding": 10,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#48BBEC",
                "fontSize": 25,
                "fontWeight": "bold",
              }
            }>
            4
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#656565",
                "fontSize": 20,
              }
            }>
            Charmander
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                undefined,
              ]
            }
            value={false} />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ddd",
            "height": 1,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "padding": 10,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#48BBEC",
                "fontSize": 25,
                "fontWeight": "bold",
              }
            }>
            5
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#656565",
                "fontSize": 20,
              }
            }>
            Charmeleon
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                undefined,
              ]
            }
            value={false} />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ddd",
            "height": 1,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "padding": 10,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#48BBEC",
                "fontSize": 25,
                "fontWeight": "bold",
              }
            }>
            6
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#656565",
                "fontSize": 20,
              }
            }>
            Charizard
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                undefined,
              ]
            }
            value={false} />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ddd",
            "height": 1,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "padding": 10,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#48BBEC",
                "fontSize": 25,
                "fontWeight": "bold",
              }
            }>
            7
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#656565",
                "fontSize": 20,
              }
            }>
            Squirtle
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                undefined,
              ]
            }
            value={false} />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ddd",
            "height": 1,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "padding": 10,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#48BBEC",
                "fontSize": 25,
                "fontWeight": "bold",
              }
            }>
            8
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#656565",
                "fontSize": 20,
              }
            }>
            Wartortle
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                undefined,
              ]
            }
            value={false} />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ddd",
            "height": 1,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "padding": 10,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#48BBEC",
                "fontSize": 25,
                "fontWeight": "bold",
              }
            }>
            9
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#656565",
                "fontSize": 20,
              }
            }>
            Blastoise
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                undefined,
              ]
            }
            value={false} />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ddd",
            "height": 1,
          }
        } />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "padding": 10,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#48BBEC",
                "fontSize": 25,
                "fontWeight": "bold",
              }
            }>
            10
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#656565",
                "fontSize": 20,
              }
            }>
            Caterpie
          </Text>
          <RCTSwitch
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onValueChange={[Function]}
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                undefined,
              ]
            }
            value={false} />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ddd",
            "height": 1,
          }
        } />
    </View>
  </View>
</ScrollView>
`;
